### ISSUE TITLE
Child process/subprocess support for the CLI platform

### ISSUE THREAD

**Issue by ayazhafiz** [roc-lang/roc#4045]
Design is here: https://rwx.notion.site/Subprocess-support-in-the-CLI-Platform-708de3aa0104437099d825937ea8905e

We are blocked on
- [ ] Implementation of the File API

before this can be implemented. We are also optionally blocked on function type support in glue, though that can be worked around incrementally.

**Comment by ayazhafiz** [comment]
Branch with a shell of the API: https://github.com/roc-lang/roc/tree/subprocess-support1

**Comment by Anton-4** [comment]
Hey @ayazhafiz, can you update the notion link so that it does not require creating an account?

I think the below procedure should do it:
> To share a Notion page without login, select Share at the top, then click the toggle next to Share to Web. Now all you need to do is copy the URL that appears beneath it and share this with whoever you want to have access to the page.

**Comment by ayazhafiz** [comment]
Done! Thanks for the catch Anton.

**Comment by skyqrose** [comment]
The basic-cli platform is no longer part of this repo, so I'm closing the issue. If this idea is still relevant, then open a new issue in https://github.com/roc-lang/basic-cli
